src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbVacancyRepository.cs

```diff
@@ -58,6 +58,24 @@ public MongoDbVacancyRepository(IOptions<MongoDbConnectionDetails> details)
             return await result.ToListAsync();
         }
 
+        public async Task<Vacancy> GetSingleVacancyForPostcode(string postcode)
+        {
+            var builder = Builders<Vacancy>.Filter;
+            var filter = builder.Eq(v => v.EmployerLocation.Postcode, postcode) &
+                         builder.Ne(v => v.EmployerLocation.Latitude, null) &
+                         builder.Ne(v => v.EmployerLocation.Longitude, null);
+
+            var options = new FindOptions<Vacancy>
+            {
+                Sort = Builders<Vacancy>.Sort.Descending(v => v.VacancyReference),
+                Limit = 1
+            };
+
+            var collection = GetCollection<Vacancy>();
+            var result = await collection.FindAsync(filter, options);
+            return result.SingleOrDefault();
```

Comment:
Could this sequence potentially contain more than one matching vacancy with the same postcode?

Reply:
The Limit = 1 appears to limit the results